### PR TITLE
feat: switch logic to props

### PR DESCRIPTION
**Pull request by magicmatatjahu**
Switch reading block's metadata from components to the props:
- update source code
- update examples with new approach

### CODE REVIEW

File: examples/simple-react/src/social_media/System.tsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
-import React, { useState, useEffect } from 'react';
-import { AsyncAPIApplication, SystemView } from '@lagoni/edavisualiser';
-import '@lagoni/edavisualiser/styles/default.css';
-import '../simple.css';
-import '@asyncapi/parser/dist/bundle';
+import { useState, useEffect } from 'react';
+import { SystemView } from '@lagoni/edavisualiser';
+import { Menu } from './Menu';
```
Comment (jonaslagoni) [inline examples/simple-react/src/social_media/System.tsx:3]:
```suggestion
import { Menu } from './menu';
```

Comment (jonaslagoni) [inline examples/simple-react/src/social_media/System.tsx:3]:
```suggestion
import { Menu } from './menu';
```

File: examples/simple-react/src/social_media/application.tsx [hunk 2]
```diff
@@ -1,57 +1,51 @@
...
-import {
-  ApplicationFocusView,
-  AsyncAPIApplication,
-} from '@lagoni/edavisualiser';
-import '@lagoni/edavisualiser/styles/default.css';
-import '../simple.css';
-import '@asyncapi/parser/dist/bundle';
-import { apps } from './apps';
+import { useState, useEffect } from 'react';
+import { Menu } from './Menu';
```
Comment (jonaslagoni) [inline examples/simple-react/src/social_media/application.tsx:2]:
```suggestion
import { Menu } from './menu';
```

Comment (jonaslagoni) [inline examples/simple-react/src/social_media/application.tsx:2]:
```suggestion
import { Menu } from './menu';
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
For the simple react example, there is an inconsistency between lower and upper case file names that fail the examples, i.e. I cant run them 😅 There are more than the ones I found here.

For the next.js example, I am unable to run it 🤔 
```
npm run build

> build
> next build

info  - Using webpack 5. Reason: no next.config.js https://nextjs.org/docs/messages/webpack5
info  - Checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully

> Build error occurred
TypeError: Cannot read properties of undefined (reading 'registerLanguage')
    at Object.1308 (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/pages/index.js:943:16)
    at __webpack_require__ (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/webpack-runtime.js:25:43)
    at Object.6660 (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/pages/index.js:82:16)
    at __webpack_require__ (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/webpack-runtime.js:25:43)
    at Object.9815 (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/pages/index.js:819:18)
    at __webpack_require__ (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/webpack-runtime.js:25:43)
    at Object.5298 (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/pages/index.js:841:22)
    at __webpack_require__ (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/webpack-runtime.js:25:43)
    at Object.743 (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/pages/index.js:1604:35)
    at __webpack_require__ (/Users/lagoni/Downloads/EDAVisualiser-feat-refactor-to-props/examples/nextjs/.next/server/webpack-runtime.js:25:43) {
  type: 'TypeError'
}
info  - Collecting page data .% 
```

**Comment by magicmatatjahu** [comment]
@jonaslagoni Hmm, strange with build. It's my logs after run `npm run build`. Could you try to remove `node_modules` and install deps from scratch?

image

Also regarding to the file names. I changed git config, after last commit everything should be fine :)

**Review by jonaslagoni — APPROVED** [review]
> @jonaslagoni Hmm, strange with build. It's my logs after run `npm run build`. Could you try to remove `node_modules` and install deps from scratch?

Hmm, might be my setup then 🤔 But I have the same issue as before.

Lets merge this as the simple-react works fine, and is a baseline for the current setup 👍 

Thanks for taking the time to drive this!

**Pull request merged**